### PR TITLE
feat: Enable Jetpack collection for native block inserter

### PR DESCRIPTION
**Pull request by dcalhoun**
Display a collection of Jetpack blocks within the native block inserter to increase brand awareness during the Jetpack app migration. 

#### Changes proposed in this Pull Request:

* The web editor displays a collection of Jetpack blocks in the block inserter for non-WordPress.com, Jetpack-connected sites. This change enables that same collection display for the native block inserter. 
* Rename the collection from "Jetpack" to "Jetpack powered" for the native block inserter. 
* Add a separate `block-category.native.js` file to configure native platform without adding additional code/complexity to the web bundle. 

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

 
p9ugOq-33V-p2

#### Does this pull request change what data or activity we track or use?

No. 

#### Testing instructions:

See https://github.com/wordpress-mobile/gutenberg-mobile/pull/5024.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/shared/block-category.native.js
Comment (dcalhoun) [inline projects/plugins/jetpack/extensions/shared/block-category.native.js:8]: There are only two differences between this native file and the sibling web file: 

1. The [`isWpCom` conditional](https://github.com/Automattic/jetpack/blob/52dc9943f51d013cd1405e0e8f4d66c1326f729b/projects/plugins/jetpack/extensions/shared/block-category.js#L9) is removed (this check relies upon `window` and does not work for the native editor). 
2. Display `JetpackPoweredBadge` rather than `JetpackLogo`. 

A separate native file feels a bit excessive for such small differences, but it is likely for the best so that (1) the web bundle size does not increase with native code and (2) a misleading `isWpCom` no-op is not run for the native editor.

File: projects/plugins/jetpack/extensions/shared/block-category.native.js
Comment (dcalhoun) [inline projects/plugins/jetpack/extensions/shared/block-category.native.js]: There are only two differences between this native file and the sibling web file: 

1. The [`isWpCom` conditional](https://github.com/Automattic/jetpack/blob/52dc9943f51d013cd1405e0e8f4d66c1326f729b/projects/plugins/jetpack/extensions/shared/block-category.js#L9) is removed (this check relies upon `window` and does not work for the native editor). 
2. Changing the collection title from "Jetpack" to "Jetpack powered." 

A separate native file feels a bit excessive for such small differences, but it is likely for the best so that (1) the web bundle size does not increase with native code and (2) a misleading `isWpCom` no-op is not run for the native editor.

Comment (fluiddot) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: @dcalhoun I'm wondering if this should only be enabled for the WordPress app, probably showing the Jetpack logo in the Jetpack app would be a bit redundant, wdyt?

Comment (SiobhyB) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: +1 to this seeming a bit redundant for the Jetpack app, but not necessarily a blocker to merging if limiting it to only the WP app proves to be complex (especially as the eventual plan is to remove it).

Comment (dcalhoun) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: That is a great thought. I agree displaying "Jetpack powered" within the Jetpack app may be redundant. 

@fluiddot @SiobhyB what approach do you think might be appropriate for controlling this? 

We would need to set/access a value indicating the WordPress/Jetpack app and/or site type prior to the point we [invoke block collection registration](https://github.com/wordpress-mobile/gutenberg-mobile/blob/c7c720f41877e7d60884bcc2222280be3860c688/src/jetpack-editor-setup.js#L89). 

I perceive the `capabilities` to be of similar nature, but they target individual features, and do not describe a higher-level app/site "type." Relatedly, the Jetpack plugin houses a set of [utilities for determining site type](https://github.com/Automattic/jetpack/blob/fe810ca5e60073b284b12cd2e2f5a3d07674a9f1/projects/js-packages/shared-extension-utils/src/site-type-utils.js#L8-L12), but they are incompatible with the native context. 

We could possibly check each of the [Jetpack block capabilities](https://github.com/wordpress-mobile/gutenberg-mobile/blob/c7c720f41877e7d60884bcc2222280be3860c688/src/jetpack-editor-setup.js#L75-L86)...

```js
if (mediaFilesCollectionBlock || contactInfoBlock || tiledGalleryBlock) {
  // register Jetpack block collection
}
```

...or add a new Jetpack site/app capability. 

```js
if (capabilities.jetpack) {
  // register Jetpack block collection
}
```

Are there alternative approaches I am overlooking?

Comment (fluiddot) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > +1 to this seeming a bit redundant for the Jetpack app, but not necessarily a blocker to merging if limiting it to only the WP app proves to be complex (especially as the eventual plan is to remove it).

Good point, I also think that only enabling it for the WP app shouldn't be a blocker.

Comment (fluiddot) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > Are there alternative approaches I am overlooking?

I noticed that in the iOS native side we're using an object named [`AppConfiguration`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) that contains different flags, including `isJetpack` to determine if the app is the Jetpack one. As an example, this flag is being used for determining when the Jetpack banner should be displayed ([reference](https://github.com/wordpress-mobile/WordPress-iOS/blob/ddb0ba2c7e48e6e3989bc8c948592f71da7b4344/WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift#L606-L608)), so we could use that value for this too.

In relation to how we could pass this information to RN side, I was thinking to have a new capability or initial prop to identify the client. This way we could enable/disable the block collection depending on the client (i.e. the app), wdyt?

Comment (dcalhoun) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > I noticed that in the iOS native side we're using an object named [`AppConfiguration`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) that contains different flags, including `isJetpack` to determine if the app is the Jetpack one. 

Agreed. This does seem the most appropriate flag to use. Thanks for sharing. 

> In relation to how we could pass this information to RN side, I was thinking to have a new capability or initial prop to identify the client. This way we could enable/disable the block collection depending on the client (i.e. the app), wdyt?

As I consider it further, setting the app type within `capabilities` feels a bit inappropriate. I imagine it makes more sense to have a `isJetpack` boolean within the editor props/settings. 

I can begin work to limit when the "Jetpack powered" badge displays, but it looks like the consensus is that work should not block merging this work. 👍🏻

Comment (fluiddot) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > As I consider it further, setting the app type within `capabilities` feels a bit inappropriate. I imagine it makes more sense to have a `isJetpack` boolean within the editor props/settings.

True, although this would imply adding it to Gutenberg and hence, add a reference to Jetpack 😅. An alternative could be to generate a different RN bundle for each WordPress and Jetpack app. This way we could set global constants similar to how it's done on the native side with [AppConfiguration](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) for each app. In fact, this option might be necessary in the future for overriding the colors of the editor for example.

> I can begin work to limit when the "Jetpack powered" badge displays, but it looks like the consensus is that work should not block merging this work. 👍🏻

Yep, let's wrap up this task first and then focus on how we can identify in the editor the app that is being run 👍 .

Comment (dcalhoun) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > True, although this would imply adding it to Gutenberg and hence, add a reference to Jetpack 😅. 

I wondered this myself. I hoped we might be able to strip out the value within `gutenberg-mobile`'s editor setup code, but we may still run into issue where the new property may have to be defined within `react-native-bridge` protocols. 

> An alternative could be to generate a different RN bundle for each WordPress and Jetpack app. This way we could set global constants similar to how it's done on the native side with [AppConfiguration](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) for each app. In fact, this option might be necessary in the future for overriding the colors of the editor for example.

Another alternative might be to define a generic host app identifier, e.g. `hostAppNamespace: 'jetpack'|'wordpress'|'awesome-future-app'`.

**Review by SiobhyB — DISMISSED** [review]
Approving following the review at https://github.com/WordPress/gutenberg/pull/42405#pullrequestreview-1040408602, with the understanding there will likely be [a follow up task](https://github.com/Automattic/jetpack/pull/25092/files#r921724500) to limit this change to the WP app. 

In addition, we'll need to get approval from the Jetpack team before this PR's merged (the `[Status] Needs Review` label can be used to get this final review).

File: projects/plugins/jetpack/extensions/shared/block-category.native.js
Comment (fluiddot) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: > Another alternative might be to define a generic host app identifier, e.g. `hostAppNamespace: 'jetpack'|'wordpress'|'awesome-future-app'`.

Yeah, this is aligned to what I [commented](https://github.com/Automattic/jetpack/pull/25092#discussion_r922205803) about passing the editor's client, although I agree that `hostAppNamespace` sounds more accurate.

**Comment by dcalhoun** [comment]
Thank you, Siobhan! Marking this as ready for review by the Jetpack crew now. 

Note for the crew: I don't have commit access for WP.com, so would be grateful if you could handle committing to WP.com for me too. 🙇🏻

File: projects/plugins/jetpack/extensions/shared/block-category.native.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/shared/block-category.native.js]: Should this be translated, like the other titles?

```suggestion
	title: __( 'Jetpack powered', 'jetpack' ),
```

Comment (dcalhoun) [reply projects/plugins/jetpack/extensions/shared/block-category.native.js]: Yes. Thank you for catching that!

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
r249105-wpcom

**Pull request merged**
